services/src/main/java/org/keycloak/services/resources/admin/AdminRoot.java

```diff
@@ -97,6 +102,11 @@ public static UriBuilder adminBaseUrl(UriBuilder base) {
      */
     @GET
     public Response masterRealmAdminConsoleRedirect() {
+
+        if (!ADMIN_CONSOLE_ENABLED) {
```

Comment:
It should be fine to use `Profile.isFeatureEnabled(Profile.Feature.ADMIN2) || Profile.isFeatureEnabled(Profile.Feature.ADMIN)` directly here.

Quarkus applications have a well-defined initialization process and using `Profile` in static init might cause issues as they are executed most of the time when the class is loaded.

Reply:
Good point I'll adapt the code